### ISSUE TITLE
test test_read_encrypted_file fails when $GNUPGHOME is not set

### ISSUE THREAD

**Issue by blais** [beancount/beancount#263]
**[Original report](https://bitbucket.org/blais/beancount/issue/263) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

I get:

```
ERROR: test_read_encrypted_file (beancount.utils.encryption_test.TestEncryptedFiles)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tbm/beancount/beancount/utils/encryption_test.py", line 132, in test_read_encrypted_file
    self.assertEqual(INPUT, plaintext)
  File "/usr/lib/python3.6/contextlib.py", line 88, in __exit__
    next(self.gen)
  File "/home/tbm/beancount/beancount/utils/test_utils.py", line 359, in environ
    os.environ[varname] = oldvalue
  File "/usr/lib/python3.6/os.py", line 674, in __setitem__
    value = self.encodevalue(value)
  File "/usr/lib/python3.6/os.py", line 744, in encode
    raise TypeError("str expected, not %s" % type(value).__name__)
TypeError: str expected, not NoneType
```

I don't have `$GNUPGHOME` defined and it barfs when it tries to set the old value after the test (since there's no old value).

I'll propose a PR.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Merged in tbm13/beancount/bug263 (pull request #66)

Fixed #263: Only restore env variable in environ() if variable existed before

Approved-by: Martin Blais <blais@furius.ca>

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #263: Only restore env variable in environ() if variable existed before

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Merged in tbm13/beancount/bug263 (pull request #66)

Fixed #263: Only restore env variable in environ() if variable existed before

Approved-by: Martin Blais <blais@furius.ca>
